src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -3311,6 +3362,17 @@ static void updateEntity
     toUnset.append(ENT_LOCATION, 1);
   }
 
+  // We don't touch toSet in the replace case, due to
+  // the way in which BSON is composed in that case (see below)
+  if ((currentDateExpiration != NO_EXPIRATION_DATE) && (action != ActionTypeReplace))
```

Comment:
The effect should not be the same. In this way, if  the action is ActionTypeReplace and dateExpires attribute is not in the payload (then replaceDateExpiration is false) , it will not append this attribute to those to be unset, as the replace should do.

Reply:
Just to check if I'm understand correctly...

Do you mean that with my proposal if action type is replaced and dateExpires is not in the payload then unset ENT_EXPIRATION is not added to the unset array?

(Btw, reading your comment maybe a better name for replaceDateExpiration around the code is dateExpirationInPayload. Just a suggestion... replaceDateExpiration is also fine)

Reply:
Exactly, I mean this. Anyway, dateExpirationInPayload sounds better than replaceDateExpiration :)

Reply:
What about using

```
```
if (action != ActionTypeReplace) 
{
  if (currentDateExpiration != NO_EXPIRATION_DATE)
  {
  ..
  }
  else if (!replaceDateExpiration)
  {
  ...
  }
}
```

and, at the same time to use within the `if (action == ActionTypeReplace)` below:

```
    if (replaceDateExpiration)
    {
         ...
    else
    {
      updatedEntity.append("$set", BSON(ENT_ATTRS                   << toSetObj <<
                                        ENT_ATTRNAMES               << toPushArr <<
                                        ENT_MODIFICATION_DATE       << now <<
                                        ENT_LAST_CORRELATOR         << fiwareCorrelator));
      updatedEntity.append("$unset", BSON(ENT_EXPIRATION <<1));
    }

```

?

I mean, simplify the first part and be explicit with $unset in the second part.

(Again, just a suggestion :) Unfortunatelly, the updateEntity() is tricky and none of the solutions has a clear advantage in simplification terms over the other. Please, choose the one you prefer either keeping the existing one in the PR or what I propose, if it correct).

Reply:
By using
```updatedEntity.append("$unset", BSON(ENT_EXPIRATION <<1));```
We lose other values appended in "toUnset" previously, such as:
```toUnset.append(ENT_LOCATION, 1);```

I would keep everything as is :)

Reply:
Ok :)

Thus, only dateExpirationInPayload  is pending with regard this comment thread, isn't it?

Reply:
Yes, already fixed. Once testing is finished, I will push it.

Reply:
Fixed in 9b3853d